ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/ConfigManagementServiceIT.scala

```diff
@@ -83,4 +83,83 @@ final class ConfigManagementServiceIT extends LedgerTestSuite {
   // TODO(JM): Test that sets the time model and verifies that a transaction with invalid
   // ttl/mrt won't be accepted. Can only implement once ApiSubmissionService properly
   // uses currently set configuration.
+
+  test(
+    "CMSetConflicting",
+    "Conflicting generation should be rejected",
+    allocate(NoParties),
+    runConcurrently = false,
+  )(implicit ec => { case Participants(Participant(ledger)) =>
+    for {
+      // Get the current time model
+      response1 <- ledger.getTimeModel()
+      oldTimeModel = {
+        assert(response1.timeModel.isDefined, "Expected time model to be defined")
+        response1.timeModel.get
+      }
+
+      // Set a new time model with the next generation
+      t1 <- ledger.time()
+      _ <- ledger.setTimeModel(
+        mrt = t1.plusSeconds(30),
+        generation = response1.configurationGeneration,
+        newTimeModel = oldTimeModel,
+      )
+
+      // Set a new time model with the same generation
+      t2 <- ledger.time()
+      failure <- ledger
+        .setTimeModel(
+          mrt = t2.plusSeconds(30),
+          generation = response1.configurationGeneration,
+          newTimeModel = oldTimeModel,
+        )
+        .mustFail("setting Time Model with an outdated generation")
+    } yield {
+      assertGrpcError(failure, Status.Code.INVALID_ARGUMENT, "")
```

Comment:
@mziolekda  pls not here it is an INVALID_ARGUMENT, I think this is ok, probably in the command service the current config is read, and we throw it away there already